### PR TITLE
Bugfix for Issue #38599 - Smart Search indexing macros from articles

### PR DESCRIPTION
**Pull request by MacJoom**
Pull Request for Issue #38599 .
finder aka smart search index contains macros of loadmodule in search results
### Summary of Changes
in onContentPrepare of loadmodule:
while indexing remove macros (loadposition, loadmoduleid, loadmodule) from article text with preg_replace

### Testing Instructions (from #38599)

Enable smart search aka finder and publish a finder search module somewhere or connect a menu item to the smart search item type.
Enable the load modules plugin.
Enable the smart search content plugin.
Create a site module of type custom with some content and place it at custom positon herewego.
Create an article and place a loadposition macro in that article like this:
{loadposition herewego}

Now the finder automatically indexes the new article.
Go to the search box and search for loadposition or herewego.

Within the Smart Search component you can clear the index under maintenace and reindex with the index button
Do a clear before applying the PR. 

### Actual result BEFORE applying this Pull Request

I would not expect to see search results containing the article with the loadposition macro.

e.g. {loadposition herewego} in the search results

I would not expect the search suggestions list containing the search items loadposition and herewego.
Instead I would expect that the macro is deleted in this case before indexing.

### Expected result AFTER applying this Pull Request

The search result does contain the article where the loadposition macro is placed and the suggestion list does not contain both loadposition and herewego.

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [x] No documentation changes for manual.joomla.org needed

### CODE REVIEW

**Comment by Quy** [comment]
Please fix conflicts. Thanks.

**Comment by MacJoom** [comment]
I don't know why _modals.css is in this commit... - can be removed

File: plugins/content/loadmodule/loadmodule.php [hunk 1]
```diff
@@ -69,6 +64,24 @@ public function onContentPrepare($context, &$article, &$params, $page = 0)
 
         // Expression to search for(id)
         $regexmodid = '/{loadmoduleid\s([1-9][0-9]*)}/i';
+        // Remove macros and don't run this plugin when the content is being indexed
+
+        if ($context === 'com_finder.indexer') {
+            //Remove macros if any
+            if (strpos($article->text, 'loadposition')) {
```
Comment (Quy) [inline plugins/content/loadmodule/loadmodule.php]:
```suggestion

        // Remove macros and don't run this plugin when the content is being indexed
        if ($context === 'com_finder.indexer') {
            if (strpos($article->text, 'loadposition')) {
```

Comment (Quy) [reply plugins/content/loadmodule/loadmodule.php]: Use `str_contains` like below?

Comment (MacJoom) [reply plugins/content/loadmodule/loadmodule.php]: I just looked up str_contains and it was introduced in PHP 8.0 - so it shouldnt be used for Joomla 4 - i have to test what happens when running PHP 7. So the later usage of str_contains may be wrong

Comment (MacJoom) [reply plugins/content/loadmodule/loadmodule.php]: Ah i see it's done with symfony polyfill - then i can change it

File: plugins/content/loadmodule/loadmodule.php [hunk 2]
```diff
@@ -69,6 +64,24 @@ public function onContentPrepare($context, &$article, &$params, $page = 0)
 
         // Expression to search for(id)
         $regexmodid = '/{loadmoduleid\s([1-9][0-9]*)}/i';
+        // Remove macros and don't run this plugin when the content is being indexed
+
+        if ($context === 'com_finder.indexer') {
+            //Remove macros if any
```
Comment (Quy) [inline plugins/content/loadmodule/loadmodule.php]:
```suggestion

        // Remove macros and don't run this plugin when the content is being indexed
        if ($context === 'com_finder.indexer') {
```

**Comment by ChrisHoefliger** [comment]
I have tested this item :white_check_mark: successfully on 9a37f4e3c45256b2183bf8c683c946337b84ed1d<br /><br />Tested on Joomla 4.3.4 @ PBF 2023<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38998">issues.joomla.org/tracker/joomla-cms/38998</a>.</sub>

**Comment by rachelwalraven** [comment]
I have tested this item :white_check_mark: successfully on 9a37f4e3c45256b2183bf8c683c946337b84ed1d<br /><br />On Joomla 4.3 with PHP 8.1.22<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38998">issues.joomla.org/tracker/joomla-cms/38998</a>.</sub>

**Comment by alikon** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38998">issues.joomla.org/tracker/joomla-cms/38998</a>.</sub>

**Comment by obuisard** [comment]
Thank you Martin @MacJoom for this PR!

**Pull request merged**